pages/slice/language-guide/class-types.md

```diff
@@ -178,6 +178,25 @@ When the generated code decodes a class in sliced format and slices off derived
 slices: it keeps them in the decoded instance. If you later send this class instance to another application that knows
 the derived class type, this application will decode successfully the full type.
 
+## Compact type IDs
+
+When the generated code encodes a class it needs to transmit the class's type.
+It does this by encoding the class's _type ID_ - a unique value associated with each class.
+Normally, this is the class's fully-qualified identifier.
+However, to reduce overhead, you can assign a numeric type ID to a class instead. For example:
+
+```slice
+module MyModule
+{
+    class ClassWithStringTypeId {}     // has a type ID of "MyModule::ClassWithStringTypeId"
+    class ClassWithNumericTypeId(9) {} // has a type ID of `9`
```

Comment:
That's incorrect. 

The type ID of ClassWithNumericTypeId is "::MyModule:...". But this class has a compact type ID _in addition_ to its type ID.

Reply:
(getting into the encoding weeds here)

If you add a compact type ID to a class, you can still decode instances encoded without this compact type ID (they include the regular type ID that you know). However, if you encode a class with a compact type ID, the recipient has to know this compact type ID - else decoding failure.

Reply:
Yes, everything you said is correct, but I felt it was outside the scope of the language guide.
I'll change the 2nd line to "compact type ID" for clarity.
I'm not sure if there's anything else you're suggesting I should add.